src/app/dao/tests/PartDAO.spec.ts

```diff
@@ -60,43 +84,363 @@ describe('PartDAO', () => {
+                ...PartMocks.INITIAL,
+                beginning: serverTimestamp(),
+                playerZero: creator,
+                playerOne: UserMocks.OPPONENT_MINIMAL_USER,
+            });
+
+            const updates: Partial<Part>[] = [
+                { playerZero: UserMocks.OPPONENT_MINIMAL_USER },
+                { playerOne: creator },
+                { beginning: serverTimestamp() },
+            ];
+            for (const update of updates) {
+                // When trying to change the field
+                const result: Promise<void> =
+                    partDAO.updateAndBumpIndex(partId, Player.ZERO, PartMocks.INITIAL.lastUpdate.index, update);
+                // Then it should fail
+                await expectPermissionToBeDenied(result);
+            }
+        });
+        it('should allow deleting part if it has not started', async() => {
+            // Given a non-started part and its owner (as defined in the joiner)
+            const creator: MinimalUser = await createConnectedUser(CREATOR_EMAIL, CREATOR_NAME);
+            const partId: string = await partDAO.create({ ...PartMocks.INITIAL, playerZero: creator });
+            await joinerDAO.set(partId, { ...JoinerMocks.INITIAL, creator });
+
+            // When deleting the part
+            const result: Promise<void> = partDAO.delete(partId);
+
+            // Then it should succeed
+            await expectAsync(result).toBeResolvedTo();
+        });
+        it('should forbid deleting part after it has started', async() => {
+            // Given a started part and its owner (as defined in the joiner)
+            const creator: MinimalUser = await createConnectedUser(CREATOR_EMAIL, CREATOR_NAME);
+            const partId: string = await partDAO.create({
+                ...PartMocks.INITIAL,
+                playerZero: creator,
+                beginning: serverTimestamp(),
+            });
+            await joinerDAO.set(partId, { ...JoinerMocks.INITIAL, creator });
+
+            // When deleting the part
+            const result: Promise<void> = partDAO.delete(partId);
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+    });
+    describe('for player', () => {
+        it('should forbid player to change typeGame', async() => {
+            // Given a part and a player (here, creator)
+            const playerZero: MinimalUser = await createConnectedUser(CREATOR_EMAIL, CREATOR_NAME);
+            const partId: string = await partDAO.create({ ...PartMocks.INITIAL, playerZero });
+
+            // When trying to change the game type
+            const result: Promise<void> = partDAO.updateAndBumpIndex(partId, Player.ZERO, PartMocks.INITIAL.lastUpdate.index, { typeGame: 'P4' });
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should allow to increment turn and decrement it for take backs', async() => {
```

Comment:
not as the only update, for sure !!! (or was listMoves.length not checkable from the security pov ?)

Reply:
Non y'a pas de .length sur les listes dans les règles firebase /o\

Reply:
VIVEMENT LA MIGRATION EN SUBCOLLECTION ALORS !
Tu peux resolve